src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesHttpClient.java

```diff
@@ -78,6 +83,26 @@ public Response delete(String resourceEndpoint, String appId) {
 		return response;
 	}
 
+	/**
+	 * Get API version to use for some of the Controller's API route mappings based on the kubernetes
+	 * version used in the underlying cluster.
+	 * @param client the KubernetesHttpClient to use
+	 * @return the appropriate API version to use
+	 */
+	public static String getApiVersionForK8sVersionFromCluster(KubernetesHttpClient client) {
+		try {
+			Response response = client.get("version", "");
+			Map<String, String> versionResponse = new ObjectMapper().readValue(response.body().string(), HashMap.class);
+			String k8sVersionFromCluster = versionResponse.get("gitVersion");
+			Version version = Version.valueOf(k8sVersionFromCluster.substring(1));
+			Version version110 = Version.valueOf("1.10.0");
+			return (version.greaterThanOrEqualTo(version110)) ? "v1" : "v1beta1";
```

Comment:
when testing on both GKE 1.10.0 and 1.9.6 im getting `v1beta1` back from `KubernetesHttpClient.getApiVersionForK8sVersionFromCluster(..)`. seems like i should get `v1` when running against GKE 1.10.0.

Reply:
Weird, I tested this against both 1.9.x and 1.10.0 and saw the version updated correctly. will double check

Reply:
Verified it again and it looks ok to me. What is the response you get on `/version` endpoint via HTTP request?